proxy/interceptor.go

```diff
@@ -619,6 +620,58 @@ func visitPayloads(
 				return err
 			}
 
+		case *errordetails.MultiOperationExecutionFailure:
+
```

Comment:
Some things did, in fact, get missed!

Reply:
I am not sure if the proxy ever look into error details though...

Reply:
What do you mean? It won't call with them? If so, oh well, no harm having it here anyway since they do contain payloads

Reply:
Yeah because these are part of the error

> It won't call with them? If so, oh well

Well the consequence is a payload or failure that should be encoded would not be so it is kinda important no?

Reply:
Oh, I see what you mean now.

Well that's because it's not referenced by any other proto. In fact I don't understand this at all:

```
    // ExecuteMultiOperation executes multiple operations within a single workflow.
    //
    // Operations are started atomically, meaning if *any* operation fails to be started, none are,
    // and the request fails. Upon start, the API returns only when *all* operations have a response.
    //
    // Upon failure, it returns `MultiOperationExecutionFailure` where the status code
    // equals the status code of the *first* operation that failed to be started.
    //
    // NOTE: Experimental API.
    rpc ExecuteMultiOperation (ExecuteMultiOperationRequest) returns (ExecuteMultiOperationResponse) {
```

Reply:
I think it is part of the error details https://grpc.io/docs/guides/error/#richer-error-model

Reply:
The proxy doesn't look at errors at all right now AFAIK @cretz please correct me if I am wrong

Reply:
Oh, I see, OK that makes sense. Yeah, I agree that is a possible gap. I also think probably another PR.

Reply:
> The proxy doesn't look at errors at all right now AFAIK @cretz please correct me if I am wrong

Correct, ref https://github.com/temporalio/api-go/blob/dad8b169ada911d3778e070484d1ae78a58bd22b/proxy/interceptor.go#L99-L101. The only cases I am aware of where user-provided payloads can be in an error is `ExecuteMultiOperation` and `QueryWorkflow`. I agree it is a hole in our logic in the interceptor that we should fix (walk the gRPC status details). I will open an issue. But yes we should keep them in this switch statement of course.